apps/system/sys_terminal.c

```diff
@@ -0,0 +1,58 @@
+#include "app.h"
+#include "string.h"
+
+int parse_input(char* buf, uint len) {
+    for (int i = 0, c; i < len - 1; i++) {
+        earth->tty_read((char*)&c);
+        buf[i] = (char)c;
+
+        switch (c) {
+        case 0x03:  /* Ctrl+C    */
+            buf[0] = 0;
+        case 0x0d:  /* Enter     */
+            buf[i] = 0;
+            earth->tty_write("\r\n", 2);
+            return c == 0x03? 0 : i;
+        case 0x7f:  /* Backspace */
+            c = 0;
+            if (i) earth->tty_write("\b \b", 3);
+            i = i ? i - 2 : i - 1;
+        }
+        if (c) earth->tty_write((void*)&c, 1);
+    }
+
+    buf[len - 1] = 0;
+    return len - 1;
+}
+
+int main() {
+	SUCCESS("Enter kernel process GPID_TERMINAL");
+	
+	char buf[SYSCALL_MSG_LEN];
+	strcpy(buf, "Finish GPID_TERMINAL Initialization");
+	grass->sys_send(GPID_PROCESS, buf, 36);
+
+	while (1) {
+		int sender;
```

Comment:
In general, this file seems to have a different indentation. Pleaes keep the style of 4-space indentation (not tab, not 2-space).